mention-generator/src/main/java/com/brandwatch/minibcr/mentiongenerator/service/query/QueryClient.java

```diff
@@ -0,0 +1,35 @@
+package com.brandwatch.minibcr.mentiongenerator.service.query;
+
+import java.util.List;
+
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import com.brandwatch.minibcr.mentiongenerator.model.UserQuery;
+
+@Service
+public class QueryClient {
```

Comment:
I would probably rename this 
```suggestion
public class QueryService {
```